src/_P099_XPT2046Touch.ino

```diff
@@ -0,0 +1,450 @@
+        addFormSelector(F("Events"), F("p099_events"), 6, options3, optionValues3, choice3);
+
+        {
+          P099_data_struct *P099_data = new (std::nothrow) P099_data_struct(); //static_cast<P099_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+          if (nullptr == P099_data) {
+            return success;
+          }
+          P099_data->loadTouchObjects(event->TaskIndex);
+
+          addFormSubHeader(F("Calibration"));
+
+          bool tbUseCalibration = bitRead(P099_CONFIG_FLAGS, P099_FLAGS_USE_CALIBRATION);
+          String options4[2] = { F("No"), F("Yes") };
+          int optionValues4[2] = { 0, 1 };
+          int choice4 = tbUseCalibration ? 1 : 0;
+          addFormSelector(F("Calibrate to screen resolution"), F("p099_use_calibration"), 2, options4, optionValues4, choice4, true);
+          if (tbUseCalibration) {
+            {
+              addRowLabel(F("Calibration"), F(""));
+              html_table(F(""), false);  // Sub-table
+              html_table_header(F(""));
+              html_table_header(F("x"));
+              html_table_header(F("y"));
+              html_table_header(F(""));
+              html_table_header(F("x"));
+              html_table_header(F("y"));
+            }
+            {
+              html_TR_TD();
+              addHtml(F("Top-left"));
+              html_TD();
+              addNumericBox(F("p099_cal_tl_x"), P099_data->Calibration.top_left.x, 0, 65535);
+              html_TD();
+              addNumericBox(F("p099_cal_tl_y"), P099_data->Calibration.top_left.y, 0, 65535);
+              html_TD();
+              addHtml(F("Bottom-right"));
+              html_TD();
+              addNumericBox(F("p099_cal_br_x"), P099_data->Calibration.bottom_right.x, 0, 65535);
+              html_TD();
+              addNumericBox(F("p099_cal_br_y"), P099_data->Calibration.bottom_right.y, 0, 65535);
+
+              html_end_table();
+              addFormNote(F("All x/y values must be <> 0 to enable calibration."));
+            }
+          }
+          bool bEnableCalibrationLog = bitRead(P099_CONFIG_FLAGS, P099_FLAGS_LOG_CALIBRATION);
+          addFormCheckBox(F("Enable logging for calibration"), F("p099_log_calibration"),  bEnableCalibrationLog);
+
+          addFormSubHeader(F("Touch objects"));
+
+          {
+            if (P099_CONFIG_OBJECTCOUNT > P099_MaxObjectCount) P099_CONFIG_OBJECTCOUNT = P099_MaxObjectCount;
+            byte choice5 = P099_CONFIG_OBJECTCOUNT;
+            if (choice5 == 0) { // Uninitialized, so use default
+              choice5 = P099_CONFIG_OBJECTCOUNT = P099_INIT_OBJECTCOUNT;
+            }
+            String options5[6] = { F("None"), F("8"), F("16"), F("24"), F("32"), F("40") };
+            int optionValues5[6] = { -1, 8, 16, 24, 32, 40 };
+            addFormSelector(F("# of objects"), F("p099_objectcount"), 6, options5, optionValues5, choice5, true);
+          }
+          if (P099_CONFIG_OBJECTCOUNT > -1) {
+            {

```

Comment:
Why the extra scope (the extra level of curly braces) ?
There is also an extra level at the for loop in line 216.